internal/brokers/broker_test.go

```diff
@@ -0,0 +1,291 @@
+	}{
+		"Successfully select form authentication mode":        {sessionID: "SAM_form_success"},
+		"Successfully select qrcode authentication mode":      {sessionID: "SAM_qrcode_success"},
+		"Successfully select newpassword authentication mode": {sessionID: "SAM_newpassword_success"},
+
+		// broker errors
+		"Error when selecting authentication mode": {sessionID: "SAM_error", wantErr: true},
+
+		/* Layout errors */
+
+		// Layout type errors
+		"Error when broker returns no layout":           {sessionID: "SAM_no_layout", wantErr: true},
+		"Error when broker returns invalid layout type": {sessionID: "SAM_invalid_layout_type", wantErr: true},
+
+		// Type "form" errors
+		"Error when broker returns form with no label":      {sessionID: "SAM_form_no_label", wantErr: true},
+		"Error when broker returns form with invalid entry": {sessionID: "SAM_form_invalid_entry", wantErr: true},
+		"Error when broker returns form with invalid wait":  {sessionID: "SAM_form_invalid_wait", wantErr: true},
+
+		// Type "qrcode" errors
+		"Error when broker returns qrcode with no content":   {sessionID: "SAM_qrcode_no_content", wantErr: true},
+		"Error when broker returns qrcode with invalid wait": {sessionID: "SAM_qrcode_invalid_wait", wantErr: true},
+
+		// Type "newpassword" errors
+		"Error when broker returns newpassword with no label":      {sessionID: "SAM_newpassword_no_label", wantErr: true},
+		"Error when broker returns newpassword with invalid entry": {sessionID: "SAM_newpassword_invalid_entry", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			b, _ := newBrokerForTests(t)
+
+			gotUI, err := b.SelectAuthenticationMode(context.Background(), tc.sessionID, "mode1")
+			if tc.wantErr {
+				require.Error(t, err, "SelectAuthenticationMode should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "SelectAuthenticationMode should not return an error, but did")
+
+			wantUI := testutils.LoadWithUpdateFromGoldenYAML(t, gotUI)
+			require.Equal(t, wantUI, gotUI, "SelectAuthenticationMode should return the expected mode UI, but did not")
+		})
+	}
+}
+
+func TestIsAuthorized(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		sessionID           string
+		sessionIDSecondCall string
+
+		wantAccess           string
+		wantErr              bool
+		wantAccessSecondCall string
+		wantErrSecondCall    bool
+	}{
+		"Successfully authorize":                             {sessionID: "success", wantAccess: responses.AuthAllowed},
+		"Successfully authorize after cancelling first call": {sessionID: "IA_second_call", wantAccess: responses.AuthCancelled, sessionIDSecondCall: "success", wantAccessSecondCall: responses.AuthAllowed},
+		"Denies authentication when broker times out":        {sessionID: "IA_timeout", wantAccess: responses.AuthDenied},
+
+		"Empty data gets JSON formatted": {sessionID: "IA_empty_data", wantAccess: responses.AuthAllowed},
+
```

Comment:
This is more a note than anything to change: once you get to the PR with the validation (to validate the output of the broker), please add tests here to ensure that skip has no data, access has userinfo, retry have some error messags (see the spec) and mix that with error cases asserting that we are doing the check.